sympy/polys/orthopolys.py

```diff
@@ -72,11 +72,74 @@ def dup_chebyshevt(n, K):
     """Low-level implementation of Chebyshev polynomials of the first kind."""
     if n < 1:
         return [K.one]
+    # When n is small, it is faster to directly calculate the recurrence relation.
+    if n < 64: # The threshold serves as a heuristic
+        return _dup_chebyshevt_rec(n, K)
+    return _dup_chebyshevt_prod(n, K)
+
+def _dup_chebyshevt_rec(n, K):
+    r""" Low-level implementation of Chebyshev polynomials of the first kind using a recurrence relation.
```

Comment:
I would just shorten this to `Chebyshev polynomials of the first kind using recurrence.`